pkg/operator/datasources.go

```diff
@@ -79,11 +79,12 @@ func (op *Reporting) syncReportDataSource(logger log.FieldLogger, key string) er
 		return err
 	}
 
-	return op.handleReportDataSource(logger, reportDataSource)
+	// Deep-copy otherwise we are mutating our cache
+	ds := reportDataSource.DeepCopy()
+	return op.handleReportDataSource(logger, ds)
 }
 
 func (op *Reporting) handleReportDataSource(logger log.FieldLogger, dataSource *cbTypes.ReportDataSource) error {
```

Comment:
Should all callers of this function use the deepcopy beforehand?

Reply:
This is the only caller. The handle functions generally only get called in one spot.